airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py

```diff
@@ -219,6 +219,13 @@ def transform_snowflake(config: Dict[str, Any]):
             dbt_config["oauth_client_id"] = credentials["client_id"]
             dbt_config["oauth_client_secret"] = credentials["client_secret"]
             dbt_config["token"] = credentials["refresh_token"]
+        elif credentials.get("private_key"):
```

Comment:
From what I understand it is possible for both `private_key` and `password`  fields to be configured and when that is the case, this code will always chose to use private key for auth while the code in `SnowflakeDatabase` will always chose to use username/password for auth. Is this not the case?

Reply:
The same as at destination, config object contains data just for one auth type